### PR TITLE
feat: Improve Empty component again

### PR DESCRIPTION
**Pull request by DiamondYuan**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch.
Pull request will be merged after one of collaborators approve.
Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### 👻 What's the background?

1. Describe the source of requirement, like related issue link.

2. Describe the problem and the scenario.

### 💡 Solution

1. How to fix the problem, and list final API implementation and usage sample if that is an new feature.

2. GIF or snapshot should be provided if includes UI/interactive modification.

### 📝 Changelog description

> Describe changes from userside, and list all potential break changes or other risks.

1. English description

2. Chinese description (optional)

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

**Comment by DiamondYuan** [comment]
@zombieJ  先review实现把，可以的话我再把 测试/demo 改了。

File: components/empty/index.tsx [hunk 1]
```diff
@@ -9,78 +9,77 @@ export interface TransferLocale {
...
+                let imageStyle: React.CSSProperties = {};
+
+                if (typeof image === 'string') {
                   imageNode = <img alt={alt} src={image} />;
+                } else if (!React.isValidElement(image)) {
+                  const { src, style } = image;
+                  imageNode = <img alt={alt} src={src || defaultEmptyImg} />;
+                  imageStyle = style;
+                } else {
+                  imageNode = image;
```
Comment (afc163) [inline components/empty/index.tsx]: 嵌套太深了，拆分一下 function

**Comment by DiamondYuan** [comment]
ImageType 这个名字是否需要改成 EmptyImage

File: components/empty/__tests__/index.test.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ import Empty from '..';
 
 describe('Empty', () => {
   it('image size should change', () => {
-    const wrapper = mount(<Empty imageSize={20} />);
+    const wrapper = mount(<Empty image={{ style: { height: '20px' } }} />);
```
Comment (afc163) [inline components/empty/__tests__/index.test.js]: 还是暴露一个 `imageStyle` 吧，保持 `image` 类型为 string。

File: components/empty/index.zh-CN.md [hunk 3]
```diff
@@ -23,4 +23,5 @@ cols: 1
 | 参数 | 说明 | 类型 | 默认值 |
 | --- | --- | --- | --- |
 | description | 自定义描述内容 | string \| ReactNode | - |
-| image | 设置显示图片，为 string 时表示自定义图片地址。在 3.16.0 后，可以设置为 `Empty.PRESENTED_IMAGE_SIMPLE` 选择另一种风格的内置图片 | string \|{ src:string, style:React.CSSProperties }\| ReactNode | default |
+| imageStyle | 图片样式 | 同 style | - |
+| image | 设置显示图片，为 string 时表示自定义图片地址。在 3.16.0 后，可以设置为 `Empty.PRESENTED_IMAGE_SIMPLE` 选择另一种风格的内置图片 | string \| ReactNode | `Empty.PRESENTED_IMAGE_DEFAULT` |
```
Comment (afc163) [inline components/empty/index.zh-CN.md]: > 在 3.16.0 后，可以设置为 `Empty.PRESENTED_IMAGE_SIMPLE` 选择另一种风格的内置图片

这句可以移到外面了，`Empty.PRESENTED_IMAGE_SIMPLE` 是内建图片的引用。

File: components/empty/__tests__/index.test.js [hunk 4]
```diff
@@ -4,7 +4,7 @@ import Empty from '..';
 
 describe('Empty', () => {
   it('image size should change', () => {
-    const wrapper = mount(<Empty imageSize={20} />);
+    const wrapper = mount(<Empty imageStyle={{ height: '20px' }} />);
```
Comment (zombieJ) [inline components/empty/__tests__/index.test.js]: `{ height: '20px' }` => `{ height: 20 }`

File: components/empty/demo/customize.md [hunk 5]
```diff
@@ -19,7 +19,9 @@ import { Empty, Button } from 'antd';
 ReactDOM.render(
   <Empty
     image="https://gw.alipayobjects.com/mdn/miniapp_social/afts/img/A*pevERLJC9v0AAAAAAAAAAABjAQAAAQ/original"
-    imageSize={200}
+    imageStyle={{
+        height: '200px',
+    }}
```
Comment (zombieJ) [inline components/empty/demo/customize.md]: => `height: 200`

Comment (afc163) [reply components/empty/demo/customize.md]: wrong indent

File: components/empty/index.zh-CN.md [hunk 6]
```diff
@@ -12,6 +12,12 @@ cols: 1
 
 当目前没有数据时，用于显式的用户提示。
 
+## 内置图片 (3.16.0 以上版本)
+
++ Empty.PRESENTED_IMAGE_SIMPLE
+
++ Empty.PRESENTED_IMAGE_DEFAULT
```
Comment (afc163) [inline components/empty/index.zh-CN.md]: 把图片 `` 贴出来好了。

File: components/empty/demo/simple.md [hunk 7]
```diff
@@ -7,17 +7,17 @@ title:
...
 
-You can choose another style of image by setting image to 'simple'.
+You can choose another style of image by setting image to Empty.PRESENTED_IMAGE_SIMPLE
 
 ```jsx
 import { Empty } from 'antd';
 
 ReactDOM.render(
-  <Empty image="simple" imageSize={40} />,
+  <Empty image={Empty.PRESENTED_IMAGE_SIMPLE} imageStyle={{height:40}} />,
```
Comment (zombieJ) [inline components/empty/demo/simple.md]:
```diff
- {height:40}
+ { height: 40 }
```

File: components/empty/index.en-US.md [hunk 8]
```diff
@@ -11,6 +11,13 @@ Empty state placeholder.
 
 When there is no data provided, display for friendly tips.
 
+
+## Built-in image
+
++ Empty.PRESENTED_IMAGE_SIMPLE
```
Comment (zombieJ) [inline components/empty/index.en-US.md]: 加个图示？

File: components/empty/index.en-US.md [hunk 9]
```diff
@@ -22,5 +29,5 @@ When there is no data provided, display for friendly tips.
 | Property | Description | Type | Default |
 | -------- | ----------- | ---- | ------- |
 | description | Customize description | string \| ReactNode | - |
-| imageSize | image size | number | - |
-| image | Customize image. Will tread as image url when string provided. since 3.16.0,you can choose another style of image by setting image to 'simple'. | string \| ReactNode | false |
+| imageStyle | style of image | same as style | - |
```
Comment (zombieJ) [inline components/empty/index.en-US.md]: `same as style` => `CSSProperties`

**Comment by DiamondYuan** [comment]
用 base64 ？

**Comment by afc163** [comment]
image

image

文档里直接贴链接展示就行。

File: components/empty/index.tsx [hunk 10]
```diff
@@ -9,78 +9,69 @@ export interface TransferLocale {
...
-        className,
-        prefixCls: customizePrefixCls,
-        image = 'default',
-        description,
-        imageSize,
-        children,
-        ...restProps
-      } = props;
-      const prefixCls = getPrefixCls('empty', customizePrefixCls);
+export default class Empty extends React.Component<EmptyProps, any> {
```
Comment (zombieJ) [inline components/empty/index.tsx]: 参考一下 Typography 的写法，还是用 function component：https://github.com/ant-design/ant-design/blob/master/components/typography/index.tsx

Comment (zombieJ) [inline components/empty/index.tsx]: 参考一下 Typography 写法，还是可以继续用 function component 的：https://github.com/ant-design/ant-design/blob/master/components/typography/index.tsx

File: components/empty/index.en-US.md [hunk 11]
```diff
@@ -16,8 +16,13 @@ When there is no data provided, display for friendly tips.
 
 + Empty.PRESENTED_IMAGE_SIMPLE
 
+![image](https://user-images.githubusercontent.com/507615/54591679-b0ceb580-4a65-11e9-925c-ad15b4eae93d.png)
```
Comment (afc163) [inline components/empty/index.en-US.md]: 用 `` 吧，可以约束一下大小。

File: components/empty/index.en-US.md [hunk 12]
```diff
@@ -11,6 +11,18 @@ Empty state placeholder.
...
+## Built-in image
+
++ Empty.PRESENTED_IMAGE_SIMPLE
+
+<img src="https://user-images.githubusercontent.com/507615/54591679-b0ceb580-4a65-11e9-925c-ad15b4eae93d.png" width="100px">
+
+
++ Empty.PRESENTED_IMAGE_DEFAULT
+
+<img src="https://user-images.githubusercontent.com/507615/54591670-ac0a0180-4a65-11e9-846c-e55ffce0fe7b.png" width="100px">
```
Comment (afc163) [inline components/empty/index.en-US.md]: 大小按目前组件内的来吧。

**Review by zombieJ — APPROVED** [review]

**Comment by afc163** [comment]
#15442

**Pull request merged**